File: GradProject/FreeSpace.Web/Models/CommentModel.cs

namespace FreeSpace.Web.Models
{
    public class CommentModel
    {
        public Guid UserId { get; set; }
        public Guid PostId { get; set; }
        public Guid CommentId { get; set; }

        public string Content { get; set; }
        public string? FullName { get; set; }
        public string? ProfilePicture { get; set; }
        public DateTimeOffset? CreatedDate { get; set; }
        public List<CommentLikeModel>? CommentLikes { get; set; }

        public bool? IsLiked { get; set; }
        public int? LikesCount { get; set; }


    }
}

File: GradProjectUI/src/app/models/comment-model.ts
export class CommentModel {
  postId: any;
  commentId: any;
  content: string;
  profilePicture: string;
  fullName: string;
  createdDate: Date;
  likesCount: number;
  isLiked: boolean;

  constructor() {
    this.postId = null; // Initialize postId to null or appropriate value
    this.content = ''; // Initialize content to empty string
    this.profilePicture = ''; // Initialize profilePicture to empty string
    this.fullName = ''; // Initialize fullName to empty string
    this.createdDate = new Date(); // Initialize createdDate to current date/time
    this.likesCount = 0; // Initialize likesCount to 0
    this.isLiked = false; // Initialize isLiked to false
  }
}
